k8s/consul/Makefile

```diff
@@ -3,34 +3,53 @@ include ../gcloud.Makefile
 include ../var.Makefile
 include ../images.Makefile
 
-APP_ID ?= consul
-
-SOURCE_REGISTRY ?= marketplace.gcr.io/google
+CHART_NAME ?= consul
+APP_ID ?= $(CHART_NAME)
 
 TRACK ?= 1.4
+CONSUL_EXPORTER_TAG ?= exporter
 METRICS_EXPORTER_TAG ?= v0.5.1
 
-# List of images used in application
-TARGET_IMAGES := consul-exporter prometheus-to-sd
+SOURCE_REGISTRY ?= marketplace.gcr.io/google
+IMAGE_CONSUL ?= $(SOURCE_REGISTRY)/consul1:$(TRACK)
+IMAGE_CONSUL_EXPORTER ?= $(SOURCE_REGISTRY)/consul1:$(CONSUL_EXPORTER_TAG)
+IMAGE_PROMETHEUS_TO_SD ?= k8s.gcr.io/prometheus-to-sd:$(METRICS_EXPORTER_TAG)
+
 
-# Image variable names should correspond with TARGET_IMAGES list
-image-$(APP_ID) := $(call get_sha256,$(SOURCE_REGISTRY)/consul1:$(TRACK))
-image-consul-exporter := $(call get_sha256,$(SOURCE_REGISTRY)/consul1:exporter)
-image-prometheus-to-sd := $(call get_sha256,k8s.gcr.io/prometheus-to-sd:$(METRICS_EXPORTER_TAG))
+# Main image
+image-$(CHART_NAME) := $(call get_sha256, $(IMAGE_CONSUL))
 
-# TODO(aav66): Replace with C2D_RELEASE variable after container is updated.
-C2D_CONTAINER_RELEASE := $(call get_var_from_container,$(image-$(APP_ID)),CONSUL_VERSION)
+# List of images used in application
+ADDITIONAL_IMAGES := consul-exporter prometheus-to-sd
+# Additional images variable names should correspond with ADDITIONAL_IMAGES list
+image-consul-exporter :=  $(call get_sha256, $(IMAGE_CONSUL_EXPORTER))
+image-prometheus-to-sd :=  $(call get_sha256, $(IMAGE_PROMETHEUS_TO_SD))
+
+C2D_CONTAINER_RELEASE := $(call get_c2d_release,$(image-$(CHART_NAME)))
 
 BUILD_ID := $(shell date --utc +%Y%m%d-%H%M%S)
 RELEASE ?= $(C2D_CONTAINER_RELEASE)-$(BUILD_ID)
 
+# Additional variables
+ifdef METRICS_EXPORTER_ENABLED
+  METRICS_EXPORTER_ENABLED_FIELD = , "prometheusToSd.enabled": $(METRICS_EXPORTER_ENABLED)
+endif
+
 NAME ?= $(APP_ID)-1
 
 APP_PARAMETERS ?= { \
   "name": "$(NAME)", \
   "namespace": "$(NAMESPACE)" \
+  $(METRICS_EXPORTER_ENABLED_FIELD) \
 }
 
+
+app/build:: .build/$(CHART_NAME)/VERSION \
```

Comment:
No, it was not in app_v2.Makefile but in application specific Makefiles.

In previous PR it was moved to c2d_deployer.Makefile, but after some consideration we decided that it is better to return it to back. It is bad idea to have in included file rule that will be run always and you cannot override it nor add or change app components.

Especially we cannot add it to app_v2.Makefile because it will break backwards compatibility and break existing apps

Reply:
You are right, I was thinking about `c2d_deployer.Makefile` -- Sorry.

You cannot override, but because of double colon, you can use multiple targets with the same name, see https://www.gnu.org/software/make/manual/html_node/Double_002dColon.html.

And we already do it, see https://github.com/GoogleCloudPlatform/click-to-deploy/blob/667a0689873090aebf4d9817bb0042daaf6f5d1b/k8s/app_v2.Makefile#L81-L85 and https://github.com/GoogleCloudPlatform/click-to-deploy/blob/667a0689873090aebf4d9817bb0042daaf6f5d1b/k8s/zookeeper/Makefile#L43

Reply:
You're right, that is exactly the problem. What is in common file will be run always, even if we don't want it and we have no means to override it.

It is like having main function in the library, that will be run every time it's included.

Reply:
Moved common requirements to c2d_deployer.Makefile